From 7778df22c95908e772ee5064817b19c7d6707edc Mon Sep 17 00:00:00 2001
From: Phil Steitz <psteitz@apache.org>
Date: Sun, 22 Feb 2009 03:34:59 +0000
Subject: [PATCH] Reverted DelegatingConnection#close to 1.2.2 version to
 ensure open statements are closed before the underlying connection is closed.
 JIRA: DBCP-242

git-svn-id: https://svn.apache.org/repos/asf/commons/proper/dbcp/trunk@746629 13f79535-47bb-0310-9956-ffa450edef68
---
 src/java/org/apache/commons/dbcp/DelegatingConnection.java | 7 ++-----
 xdocs/changes.xml                                          | 4 ++++
 2 files changed, 6 insertions(+), 5 deletions(-)

diff --git a/src/java/org/apache/commons/dbcp/DelegatingConnection.java b/src/java/org/apache/commons/dbcp/DelegatingConnection.java
index 4564e67d1b..b91954c090 100644
--- a/src/java/org/apache/commons/dbcp/DelegatingConnection.java
+++ b/src/java/org/apache/commons/dbcp/DelegatingConnection.java
@@ -238,11 +238,8 @@ public void setDelegate(Connection c) {
      * any Statements that were not explicitly closed.
      */
     public void close() throws SQLException {
-        try {
-            _conn.close();
-        } finally {
-            _closed = true;
-        }
+        passivate();
+        _conn.close();
     }
 
     protected void handleException(SQLException e) throws SQLException {
diff --git a/xdocs/changes.xml b/xdocs/changes.xml
index 9ed003b4f1..a8dc3e1a03 100644
--- a/xdocs/changes.xml
+++ b/xdocs/changes.xml
@@ -42,6 +42,10 @@ The <action> type attribute can be add,update,fix,remove.
      new features as well as bug fixes and instrumentation.  Some bug fixes
      will change semantics (e.g. connection close will become idempotent).
      The minimum JDK level will be increased to 1.4">
+      <action dev="psteitz" type="fix" issue="DBCP-242">
+        Reverted DelegatingConnection close to 1.2.2 version to ensure
+        open statements are closed before the underlying connection is closed.
+      </action>
       <action dev="markt" type="fix">
         Add some test cases that were missing from the TestAll test suite.
       </action>
